include/SFML/Graphics/VertexBuffer.hpp

```diff
@@ -0,0 +1,378 @@
+    ///
+    ////////////////////////////////////////////////////////////
+    bool create(std::size_t vertexCount);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the vertex count
+    ///
+    /// \return Number of vertices in the vertex buffer
+    ///
+    ////////////////////////////////////////////////////////////
+    std::size_t getVertexCount() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Update the whole buffer from an array of vertices
+    ///
+    /// The \a vertex array is assumed to have the same size as
+    /// the \a created buffer.
+    ///
+    /// No additional check is performed on the size of the vertex
+    /// array, passing invalid arguments will lead to undefined
+    /// behavior.
+    ///
+    /// This function does nothing if \a vertices is null or if the
+    /// buffer was not previously created.
+    ///
+    /// \param vertices Array of vertices to copy to the buffer
+    ///
+    /// \return True if the update was successful
+    ///
+    ////////////////////////////////////////////////////////////
+    bool update(const Vertex* vertices);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Update a part of the buffer from an array of vertices
+    ///
+    /// \p offset is specified as the number of vertices to skip
+    /// from the beginning of the buffer.
+    ///
+    /// If \p offset is 0 and \p vertexCount is equal to the size of
+    /// the currently created buffer, its whole contents are replaced.
+    ///
+    /// If \p offset is 0 and \p vertexCount is greater than the
+    /// size of the currently created buffer, a new buffer is created
+    /// containing the vertex data.
+    ///
+    /// If \p offset is 0 and \p vertexCount is less than the size of
+    /// the currently created buffer, only the corresponding region
+    /// is updated.
+    ///
+    /// If \p offset is not 0 and \p offset + \p vertexCount is greater
+    /// than the size of the currently created buffer, the update fails.
+    ///
+    /// No additional check is performed on the size of the vertex
+    /// array, passing invalid arguments will lead to undefined
+    /// behavior.
+    ///
+    /// \param vertices    Array of vertices to copy to the buffer
+    /// \param vertexCount Number of vertices to copy
+    /// \param offset      Offset in the buffer to copy to
+    ///
+    /// \return True if the update was successful
+    ///
+    ////////////////////////////////////////////////////////////
+    bool update(const Vertex* vertices, std::size_t vertexCount, unsigned int offset);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Copy the contents of another buffer into this buffer
+    ///
+    /// \param vertexBuffer Vertex buffer whose contents to copy into this vertex buffer
+    ///
+    /// \return True if the copy was successful
```

Comment:
Hard to tell beforehand... As an example, if allocation of space for the copy fails because of some reason then it will fail. Reliably predicting GPU memory behaviour is almost impossible using OpenGL.